thrust/thrust/detail/type_traits.h

```diff
@@ -238,16 +241,7 @@ template<typename T>
 }; // end remove_reference
 
 template<typename T1, typename T2>
-  struct is_same
-    : public false_type
-{
-}; // end is_same
-
-template<typename T>
-  struct is_same<T,T>
-    : public true_type
-{
-}; // end is_same
+using is_same = cuda::std::is_same<T1, T2>;
```

Comment:
We need to fully qualify any type from cuda::std because thrust has a `cuda` namespace
```suggestion
using is_same = ::cuda::std::is_same<T1, T2>;
```